connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -349,30 +436,44 @@ public boolean contains(String connector) {
 
     /**
      * Write this connector configuration to persistent storage and wait until it has been acknowledged and read back by
-     * tailing the Kafka log with a consumer.
+     * tailing the Kafka log with a consumer. {@link #claimWritePrivileges()} must be successfully invoked before calling
+     * this method if the worker is configured to use a fencable producer for writes to the config topic.
      *
      * @param connector  name of the connector to write data for
      * @param properties the configuration to write
+     * @throws ProducerFencedException if write privileges were claimed by another writer before this method was invoked
+     * @throws IllegalStateException if {@link #claimWritePrivileges()} is required, but was not successfully invoked before
+     * this method was called
      */
     @Override
     public void putConnectorConfig(String connector, Map<String, String> properties) {
         log.debug("Writing connector configuration for connector '{}'", connector);
         Struct connectConfig = new Struct(CONNECTOR_CONFIGURATION_V0);
         connectConfig.put("properties", properties);
         byte[] serializedConfig = converter.fromConnectData(topic, CONNECTOR_CONFIGURATION_V0, connectConfig);
-        updateConnectorConfig(connector, serializedConfig);
+        try {
+            maybeSendFencibly(CONNECTOR_KEY(connector), serializedConfig);
+            configLog.readToEnd().get(READ_TO_END_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
```

Comment:
I realise that we're not trying to support a thread cancellation mechanism here, but shouldn't we set the thread's interrupted status when catching `InterruptedException`?

Reply:
This part isn't new, it's just an inlining of a private method that was only being used in one place. Considering it follows the logic used in all similar methods for this class and there haven't been any bugs reported related to it that I'm aware of, I'd like to err on the side of caution here and keep this logic as-is for now, though we can address in a follow-up if this is a genuine issue.